lib/tapjoy/autoscaling_bootstrap.rb

```diff
@@ -40,10 +40,29 @@ class << self
       attr_accessor :scaler_name, :config_name, :create_elb
       attr_reader :elb_name
 
+      # If you're using AutoscalingBootstrap to create a new ELB, that name goes here
       def elb_name=(str)
         @elb_name = str
       end
 
+      # If you're using AutoscalingBootstrap to join to a list of existing ELBs, that array
+      # goes here. This list can include or not include the provided elb_name, the
+      # array + a custom elb_name will be uniq-ed before being passed to Amazon
+      def elb_list=(list)
+        @elb_list = list
+      end
+
+      def elb_list
+        @elb_list ||= []
+      end
+
+      # This is the list of elbs passed to the autoscaling configuration. It will include
+      # the created elb, as well as the specific list of elbs to join. It will call uniq
+      # on the list in case you accidentally specify the same elb twice
+      def elbs_to_join
+        (elb_list + [Tapjoy::AutoscalingBootstrap.elb_name]).uniq
```

Comment:
Array o strings
On Sep 22, 2015 10:37 AM, "Ali Tayarani" <notifications@github.com> wrote:

> In lib/tapjoy/autoscaling_bootstrap.rb
> <https://github.com/Tapjoy/tass/pull/12#discussion_r40093529>:
>
> > +      # If you're using AutoscalingBootstrap to join to a list of existing ELBs, that array
> > +      # goes here. This list can include or not include the provided elb_name, the
> > +      # array + a custom elb_name will be uniq-ed before being passed to Amazon
> > +      def elb_list=(list)
> > +        @elb_list = list
> > +      end
> > +
> > +      def elb_list
> > +        @elb_list ||= []
> > +      end
> > +
> > +      # This is the list of elbs passed to the autoscaling configuration. It will include
> > +      # the created elb, as well as the specific list of elbs to join. It will call uniq
> > +      # on the list in case you accidentally specify the same elb twice
> > +      def elbs_to_join
> > +        (elb_list + [Tapjoy::AutoscalingBootstrap.elb_name]).uniq
>
> What data type is being returned? It looks cool, but I'm not sure how it
> works quite yet.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Tapjoy/tass/pull/12/files#r40093529>.
>